src/main/kotlin/com/ajax/cryptocurrency/threads/PriceSaver.kt

```diff
@@ -1,47 +1,34 @@
 package com.ajax.cryptocurrency.threads
 
+import com.ajax.cryptocurrency.config.SingleShotBackgroundJob
 import com.ajax.cryptocurrency.model.Cryptocurrency
 import com.ajax.cryptocurrency.repository.CryptocurrencyRepository
 import org.json.JSONObject
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
-import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.stereotype.Component
 import java.io.BufferedReader
 import java.io.InputStreamReader
-import java.lang.Thread.currentThread
 import java.lang.Thread.sleep
 import java.net.URL
 import java.time.OffsetDateTime
 import java.time.ZoneOffset
 
+@SingleShotBackgroundJob(startDelay = 1000, maxParallelThreads = 30)
 @Component
```

Comment:
`@SingleShotBackgroundJob` is annotated with `@Component`. So `PriceSaver` inherits `@Component` annotation. We can remove `@Component` here

Reply:
done